api/src/main/java/org/openmrs/module/queue/validators/QueueValidator.java

```diff
@@ -38,38 +35,22 @@ public void validate(Object target, Errors errors) {
 		log.debug("{}.validate", this.getClass().getName());
 		//instanceof checks for null
 		if (!(target instanceof Queue)) {
-			throw new IllegalArgumentException("The parameter target should not be null & must be of type" + Queue.class);
+			throw new IllegalArgumentException("Invalid Queue class: " + target.getClass().getName());
 		}
 		Queue queue = (Queue) target;
 		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "queue.name.null", "Queue name can't be null");
 		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "location", "queue.location.null", "Location can't be null");
 		
-		if (!isValidLocation(queue.getLocation())) {
-			errors.rejectValue("location", "queue.location.null", "Location is null or doesn't exists");
-		}
+		// TODO: Check if the location is tagged as a Queue Location?
 		
+		QueueServicesWrapper queueServices = Context.getRegisteredComponents(QueueServicesWrapper.class).get(0);
 		if (queue.getService() == null) {
 			errors.rejectValue("service", "QueueEntry.service.null", "The property service should not be null");
 		} else {
-			if (!QueueValidationUtils.isValidService(queue.getService())) {
+			if (!queueServices.getAllowedServices().contains(queue.getService())) {
 				errors.rejectValue("service", "Queue.service.invalid",
 				    "The property service should be a member of configured queue service conceptSet.");
 			}
 		}
 	}
-	
-	/**
-	 * For now checks for existence
-	 * 
-	 * @param location location to check if it exists
-	 * @return true or false if the location exists
-	 */
-	public boolean isValidLocation(@NotNull Location location) {
-		if (location == null) {
-			return false;
-		}
-		Location isExistentLocation = Context.getLocationService().getLocationByUuid(location.getUuid());
-		return isExistentLocation != null;
-	}
```

Comment:
This seemed rather pointless.  We already check for null earlier in the method, and we don't typically check whether a nested property is actually saved or not, so I removed this check.

Per my added comment above, we _do_ probably want to validate that the location is actually tagged as a Queue Location, but I'll leave that for a separate ticket if desired.